File: src/main/java/com/example/Clinic/service/RegisterRequestService.java
package com.example.Clinic.service;

import com.example.Clinic.model.RegisterRequest;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.Set;

@Service
public interface RegisterRequestService {

    public RegisterRequest addRegisterRequest(RegisterRequest request);
    public List<RegisterRequest> getAll();
    public RegisterRequest findByPatientId(Long id);
    public Optional<RegisterRequest> findById(Long id);
    public RegisterRequest update(RegisterRequest request);


}

File: src/main/webapp/reactjs/src/services/RegisterRequestService.js
import AxiosClient from "./clients/AxiosClient";

export const RegisterRequestService = {
    getAll,
    update
};



async function getAll() {
    return await AxiosClient.get(`https://localhost:8080/registerRequests`);
}

async function update(id, request) {
    return await AxiosClient.put(`https://localhost:8080/registerRequests/${id}`, request);
}
